sql/catalyst/src/main/java/org/apache/spark/sql/connector/write/Write.java

```diff
@@ -71,4 +72,14 @@ default StreamingWrite toStreaming() {
   default CustomMetric[] supportedCustomMetrics() {
     return new CustomMetric[]{};
   }
+
+  /**
+   * Return an array of custom metrics which are collected with values at the driver side only.
+   * Note that these metrics must be included in the supported custom metrics reported by
+   * `supportedCustomMetrics`.
+   */
+  default CustomTaskMetric[] reportCustomDriverMetrics() {
```

Comment:
ditto

Reply:
I have reverted this file as I plan to cover write paths in a seperate PR